src/bilder/lib/template_helpers.py

```diff
@@ -0,0 +1,133 @@
+from pathlib import Path
+
+from pyinfra.operations import files
+
+from bilder.components.hashicorp.consul_template.models import ConsulTemplateTemplate
+
+CONSUL_TEMPLATE_DIRECTORY = "/etc/consul-template"
+
+
+def place_jinja_template_file(  # noqa: WPS211
+    name: str,
+    repo_path: Path,
+    destination_path: Path,
+    context: dict,
+    watched_files: list[Path],
+    mode: str = "0644",
+    user: str = "root",
+    group: str = "root",
+):
+
+    """Interpolate and place a file on the destination at AMI bake time.
+    Predicate function that side effects multiple arguments and has no return.
+
+    :param name: The name the file will ultimately have on the destination system.
+    :type name: str
+
+    :param repo_path: The path within this code repository that contains the source template.
+    :type repo_path: Path
+
+    :param destination_path: The path on the destination system for the file to be placed at.
+        This does NOT include the filename, just the directory.
+    :type destination_path: Path
+
+    :param context: A dictionary containing the values to interpolate into the tempalte.
+    :type context: dict
+
+    :param watched_files: A list of files that will be watched on the system and used to
+        trigger service restarts when changed. This function will append the combined
+        destination_path + name to the list.
+    :type watched_files: list[Path]
+
+    :param mode: The permissions specifier for the file on the destination system. Numerical form.
+    :type mode: str
+
+    :param user: The file user of the rendered template on the system.
+    :type user: str
+
+    :param group: The group of the rendered template on the system.
+    :type group: str
+
+    :rtype: None
+    """
+    files.template(
+        name=f"Place and interpolate {name} jinja template file",
+        src=str(repo_path.joinpath(name + ".j2")),  # noqa: WPS336
+        dest=str(destination_path.joinpath(name)),
+        context=context,
+        mode="0664",
+        user=user,
+        group=group,
+    )
+    watched_files.append(destination_path.joinpath(name))
+
+
+def place_consul_template_file(  # noqa: WPS211
+    name: str,
+    repo_path: Path,
+    template_path: Path,
+    destination_path: Path,
+    consul_templates: list[ConsulTemplateTemplate],
+    watched_files: list[Path],
+    mode: str = "0664",
+    user: str = "consul-template",
+    group: str = "consul-template",
+):
```

Comment:
This might be better (from a design perspective) if it returns the `ConsulTemplateTemplate` object. The `watched_files` argument for the service watches can then just be generated as `watched_files=[template.destination for template in consul_template_config.template]`

That produces a more functional interface without implicit state mutation that can lead to "spooky action at a distance".